<launch>
  <arg name="model" default="$(find urdf_sim_tutorial)/urdf/13-diffdrive.urdf.xacro"/>
  <arg name="rvizconfig" default="$(find urdf_tutorial)/rviz/urdf.rviz" />
  
  <!--inlcude gazebo launch file -->
  <include file="$(find simulate_robot)/launch/gazebo.launch">
    <arg name="model" value="$(arg model)" />
  </include>

  <!--inlcude rviz launch file -->
  <include file="$(find simulate_robot)/launch/rviz.launch">
    <arg name="rvizconfig" value="$(arg rvizconfig)" />
  </include>
  
  <!-- load the params for the robot -->
  <rosparam command="load"
            file="$(find simulate_robot)/config/joints.yaml"
            ns="r2d2_joint_state_controller" />
  <rosparam command="load"
            file="$(find simulate_robot)/config/head.yaml"
            ns="r2d2_head_controller" />
  <rosparam command="load"
            file="$(find simulate_robot)/config/gripper.yaml"
            ns="r2d2_gripper_controller" />
  <rosparam command="load"
            file="$(find simulate_robot)/config/diffdrive.yaml"
            ns="r2d2_diff_drive_controller" />
  
  <!--spawn the robot  -->
  <node name="r2d2_controller_spawner" pkg="controller_manager" type="spawner"
    args="r2d2_joint_state_controller
          r2d2_head_controller
          r2d2_gripper_controller
          r2d2_diff_drive_controller
          --shutdown-timeout 3"/>

</launch>
